tests/events/events.go

```diff
@@ -0,0 +1,105 @@
+)
+
+type k8sObject interface {
+	metav1.Object
+	schema.ObjectKind
+}
+
+// ExpectNoEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectNoEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be not triggered")
+	expectEvent(object, eventType, reason, BeEmpty())
+}
+
+// ExpectEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be triggered")
+	expectEvent(object, eventType, reason, Not(BeEmpty()))
+}
+
+// DeleteEvents is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func DeleteEvents(object k8sObject, eventType, reason string) {
+	By("Expecting events to be removed")
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	events, err := virtClient.CoreV1().Events(namespace).List(
+		context.Background(),
+		metav1.ListOptions{
+			FieldSelector: fieldSelector,
+		},
+	)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	for _, event := range events.Items {
+		err = virtClient.CoreV1().Events(event.Namespace).Delete(context.Background(), event.Name, metav1.DeleteOptions{})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	}
+
+	EventuallyWithOffset(1, func() []k8sv1.Event {
+		events, err := virtClient.CoreV1().Events(namespace).List(context.Background(),
+			metav1.ListOptions{
+				FieldSelector: fieldSelector,
+			})
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+		return events.Items
+	}, 30*time.Second, 1*time.Second).Should(BeEmpty())
+}
+
+func expectEvent(object k8sObject, eventType, reason string, matcher types.GomegaMatcher) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(2, err).ToNot(HaveOccurred())
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	EventuallyWithOffset(2, func() []k8sv1.Event {
+		events, err := virtClient.CoreV1().Events(namespace).List(
+			context.Background(),
+			metav1.ListOptions{
+				FieldSelector: fieldSelector,
+			},
+		)
+		ExpectWithOffset(3, err).ToNot(HaveOccurred())
+		return events.Items
+	}, 30*time.Second).Should(matcher)
+}
+
+func constructFieldSelectorAndNamespace(object k8sObject, eventType, reason string) (string, string) {
+	kind := object.GroupVersionKind().Kind
+	if kind == "" {
+		kind = reflect.ValueOf(object).Type().Name()
+	}
+	kindSelector := fmt.Sprintf("involvedObject.kind=%s,", kind)
+	if kind == "" {
+		kindSelector = ""
+	}
+
+	name := object.GetName()
+	namespace := object.GetNamespace()
+
+	fieldSelector := fmt.Sprintf("%sinvolvedObject.name=%s,type=%s,reason=%s", kindSelector, name, eventType, reason)
+	_, _ = fmt.Fprintln(GinkgoWriter, fieldSelector)
```

Comment:
Is this intentionally left here? Seems a bit out of context in this format, don't you think?
If you really think it is useful for logging, perhaps it will be mere useful to log this at the callers, where some context could be given (e.g. that it is a deletion or expect).

Reply:
What do you refer to as a caller? We internally construct the selector and I used this for debugging. I think it may be useful if we happen to not match anything. (double checking we used the correct field selector)

Do you think we should move this to the `Expect` and `Eventually` additional messages?

Reply:
The callers in this case are`DeleteEvents` and `expectEvent`.
There there is context to what these logs refer to.

I usually try to avoid placing logs too deep in the call stack, because it is then hard to identify the call stack and in some cases, there may bee too many logging done (depending how often it is called).

If you can print this only when really needed, it will help troubleshooting for sure.
I guess if you try to log at an upper layer, you may be able to control when to log and when not, but I am not sure about this specific case.

Reply:
I have moved this up in the stack.